packages/wallets/solflare/src/adapter.ts

```diff
@@ -59,7 +61,9 @@ export class SolflareWalletAdapter extends BaseMessageSignerWalletAdapter {
     async connect(): Promise<void> {
         try {
             if (this.connected || this.connecting) return;
-            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+            if (this._readyState !== WalletReadyState.Loadable && this._readyState !== WalletReadyState.Installed) {
+                throw new WalletNotReadyError();
+            }
 
             if (!this._wallet) {
                 const Solflare = (await import('@solflare-wallet/sdk')).default;
```

Comment:
This should also be changed -- if you can interact with the browser extension without loading a library, please do.